### DISCUSSION TITLE
Component already registered for model

### DISCUSSION THREAD

**Discussion by Afrobeat** [Blazor-Diagrams/Blazor.Diagrams#248] [Q&A]
Hello :)
I'm currently trying to implement a custom Node based on the Documentation and the code of the example Project (DatabaseDesigner).

This is the created model
```
public class CommandActivity : NodeModel
    {
        public CommandActivity(Point position = null) : base(position, RenderLayer.HTML)
        {
            Parameter = new List<Parameters>
            {
                new Parameters
                {
                    Name = "Default"
                }
            };
        }

        public string Name { get; set; } = "CommandActivity";
}
```

And this is the corresponding Component:
```
<div class="card @(Node.Selected ? "selected" : "")" style="width: 320px;">
   <div class="card-header">
        <h6 class="mb-0">@Node.Name</h6>
    </div>
    <div>
        <a class="btn btn-primary" href="#" role="button" translate="yes">Sub-Workflow</a>
    </div>
</div>
```
When I now try to register them both with the following code inside of the OnInitialized method (CommandActivity is the Model and CommandActivityNode is the Component):
`Diagram.RegisterModelComponent< CommandActivity, CommandActivityNode>();`

I get this error:
`Error: System.Exception: Component already registered for model 'CommandActivity'.`

In my knowledge I didn't register the model before. Did anyone else had this issue or has an idea how I can fix this issue?

**Comment by zHaytam** [comment]
Hello,

It sounds like a very weird issue. Did you make sure you're not calling RegisterModelComponent anywhere else?
